### ISSUE TITLE
Consolidate onboarding, connect to Studio, settings & walkthrough

### ISSUE THREAD

**Issue by mattseddon** [iterative/vscode-dvc#3434]
There are a lot of small disparate screens related to onboarding (Setp webview) and connecting to Studio (Connect webview). There are also buttons associated with these screens in the sidebar. We also have a separate walkthrough (uses VS Code API) which users will see on first-install. The purpose of this ticket is to discuss and then implement a consolidation of this information into a single place.

My idea for an initial implementation involves reusing the layout from the plots webview. Namely this (without the ribbon):

<img width="1728" alt="image" src="https://user-images.githubusercontent.com/37993418/223903438-85987a7a-cacf-4d8d-b8d7-34f269ffd841.png">

To begin with, we can start by moving the existing sections within this layout. We can also add a new section that links to the walkthrough.

After we get everything into the same place then we can discuss the next steps. 

Please LMK your thoughts on 

- Initial section titles.
- Name for the view (Setup?).
- Whether or not we will need to expose all steps at the same time (i.e do we continue to show things like initialize Git once the action has been completed?)
- Final layout after initial implementation
- Alternatives

Note: We need to drop the code that automatically opens the walkthrough on first-install.

Related to #3331

**Comment by mattseddon** [comment]
Some next steps:

- Add a reference to the walkthrough to a new section on the page.
- Split CLI setup out from the Experiments section (add into new DVC section)
- Add a reminder of how to use DVCLive in the Setup Complete step of the Experiments section.
- Add some indicator as to whether sections are optional/completed to section titles.

**Comment by julieg18** [comment]
> Add a reference to the walkthrough to a new section on the page.

Do we have an idea on what we want this new section to consist of? Did we want something simple like:

<img width="846" alt="image" src="https://user-images.githubusercontent.com/43496356/230152526-b1d1bfbc-280d-44ae-98c7-149570917aae.png">

Or would there be more to it? 

> Add a reminder of how to use DVCLive in the Setup Complete step of the Experiments section.

When we say "reminder", are we just think of adding an extra sentence with a doc link or did you want to add more content like our dvclive framework tabs?

**Comment by mattseddon** [comment]
Note from retro: Setup should have a plots section that will help a user to get started with plots, show some snippets they can add to their `dvc.yaml` and also a button to create custom plots. Don't need to start on this until everything else is done. Could be a quick win before starting on remote setup.

**Comment by julieg18** [comment]
Working on adding DVC CLI details to our DVC Setup Section and tried a couple ideas for displaying it! Am I on the right track for what we are wanting to do or am I misunderstanding something? cc @iterative/vs-code, @shcheklein 

Original design:

<img width="825" alt="Screenshot 2023-04-19 at 6 45 49 PM" src="https://user-images.githubusercontent.com/43496356/233222887-86c86cba-f1af-4dbc-afe1-4c47bd102308.png">

Updated design ideas:

<img width="813" alt="Screenshot 2023-04-19 at 6 39 15 PM" src="https://user-images.githubusercontent.com/43496356/233222203-f259aa48-bb28-418c-b2eb-9ff9e81f1414.png">

<img width="699" alt="Screenshot 2023-04-19 at 6 38 06 PM" src="https://user-images.githubusercontent.com/43496356/233222224-476ec10a-9801-4795-b818-83584b6ca4c3.png">

Feedback I was aiming to address:
* text inside the "details" can be repeated depending on step's text
* design feels out of place or in other words, doesn't fit with the rest of the section content

**Comment by shcheklein** [comment]
@julieg18 yes, it goes in the right direction. I like the second one (the last) better. Let's pay more attention to the details.

- Location value doesn't make much sense to me at the moment. `python` - what does it mean? is it an actual location?
- Required version - `- 3` was not clear that it's a max version, is it included or not, etc
- The located ... message is not needed anymore I think (the title is already about that)
- Let's then go through all the stages of this screen (e.g. DVC is not even installed yet)

**Comment by julieg18** [comment]
> Location value doesn't make much sense to me at the moment. python - what does it mean? is it an actual location?

Apologies, should have mentioned these designs have static values. In reality, you would see something along the lines of:

`Location    /Users/juliegalvan/Desktop/workspace/onboarding-test/.env/bin/python (selected via python extension)`

Would that make more sense or do we need to make things more clear?

> Required version - - 3 was not clear that it's a max version, is it included or not, etc

Makes sense! Hmmm, maybe we could supply more of a sentence instead of just `NUMBER - NUMBER` or we could have two values, `Min Required Version: 2.52.0` and `Max Required Version: 3`. 

> The located ... message is not needed anymore I think (the title is already about that)

👍 

> Let's then go through all the stages of this screen (e.g. DVC is not even installed yet)

Will update the code based on previous comments then post how all the stages look!

**Comment by shcheklein** [comment]
> Would that make more sense or do we need to make things more clear?

We can make it more clear I think. At least it should be `/Users/juliegalvan/Desktop/workspace/onboarding-test/.env/bin` not python at then (it looks strange). 

what are the options for the `selected via python extension`?

>  we could have two values, Min Required Version: 2.52.0 and Max Required Version: 3.

it's better. Or we can add some text into the value as well.

**Comment by julieg18** [comment]
> We can make it more clear I think. At least it should be /Users/juliegalvan/Desktop/workspace/onboarding-test/.env/bin not python at then (it looks strange).

Hmm isn't using the `python` line more accurate? That is the python bin path which is what we use to run commands: 

```
/Users/juliegalvan/Desktop/workspace/onboarding-test/.env/bin/python -m dvc --version
``` 

Maybe we need to rename the "Location" to make things more clear... "Environment" or "Selected Environment"? 

> what are the options for the selected via python extension?

Currently, the options so far are:

* {LOCATION} (global)
* {LOCATION} (selected via python extension)
* {LOCATION} (selected manually)

I also considered moving this text to its own row instead of in parentheses next to the environment path.

**Comment by shcheklein** [comment]
> Maybe we need to rename the "Location" to make things more clear... "Environment" or "Selected Environment"?

Let's rename it into "Command" and show the full command, not including the `--version` and other params, wdyt?

> I also considered moving this text to its own row instead of in parentheses next to the environment path.

Tbh not sure if we need this at all (it's kinda obvious when it's global or not if we do the change above ^^). I think we need a button there in some way (can be a text link) to be able to trigger the switch.

**Comment by julieg18** [comment]
Adjusted table based off discussion! Here's the table's current format. Note that all these photos are taken in storybook, since it's easier to go through the different steps there so text arguments are static and text sizing is a little off. 

<img width="628" alt="image" src="https://user-images.githubusercontent.com/43496356/233654242-5098c028-0fd9-4ea5-baf3-f9441e017931.png">

* Wasn't sure if the "Setup Workspace" action fit best inside parentheses, but couldn't think of a good spot yet to put actions 🤔  
* Tried command instead of location! Not sure if it makes things more clear though. 

<details>
<summary>Setup Walkthrough Steps</summary>

## No Environment or DVC found

<img width="696" alt="image" src="https://user-images.githubusercontent.com/43496356/233653246-acb279ee-067d-4ef8-a952-610cec45434c.png">

## Environment Found but no DVC

<img width="774" alt="image" src="https://user-images.githubusercontent.com/43496356/233655141-573a33cc-1558-440a-a9b7-14ac297618c3.png">

I considered taking off the python path in the first sentence but wasn't sure if seeing the "Command" was clear enough to the user where packages would be installed.

## Dvc is installed

<img width="686" alt="image" src="https://user-images.githubusercontent.com/43496356/233656138-44a4a963-859e-441b-8f9f-494a66d42e8f.png">

## Dvc is the wrong version

<img width="721" alt="image" src="https://user-images.githubusercontent.com/43496356/233656924-d8ad7eb8-a743-4349-a2b1-c890453ca91f.png">

</details>

**Comment by julieg18** [comment]
Resolved most feedback except for:

> We don't need show experiments button here. Section should be collapsed by default and users should see just a summary and that status is "green". We can have an upgrade button for DVC at some point there.

I'd like to add section status in a separate pr since that will involve other sections as well. As for section collapsing, we have a related task (first item) in our tasklist but I'm not sure how to implement a none buggy solution just yet :) Could we hold this off for a separate pr as well? 

Here's how the table is looking now! 

<img width="604" alt="image" src="https://user-images.githubusercontent.com/43496356/234068205-a9791b44-d6ed-4476-81e1-dbfc6251c40c.png">

* "Select Python Interpreter" only shows if python extension is installed

<details>
<summary>Changes to "Environent Found but no DVC" step</summary>

<img width="728" alt="image" src="https://user-images.githubusercontent.com/43496356/234070526-771d436f-3e31-41bb-b198-bca99415c9db.png">

<img width="617" alt="image" src="https://user-images.githubusercontent.com/43496356/234069528-81a37498-11e0-4787-809d-aa238402abb6.png">

Unlike before, "Select Python Interpreter" is never shown even if the Python extension is available. Just "Configure", which I believe is what we wanted :) 

</details>

**Comment by shcheklein** [comment]
Feels like a bug to me (Not found, while it's actually installed):

<img width="931" alt="Screenshot 2023-04-28 at 1 41 16 PM" src="https://user-images.githubusercontent.com/3659196/235250569-14e8e2d2-e80b-40f7-b984-31b958d5a418.png">

The extension version is: 0.8.2

**Comment by mattseddon** [comment]
> Feels like a bug to me (Not found, while it's actually installed):
> 
> <img alt="Screenshot 2023-04-28 at 1 41 16 PM" width="931" src="https://user-images.githubusercontent.com/3659196/235250569-14e8e2d2-e80b-40f7-b984-31b958d5a418.png">
> 
> The extension version is: 0.8.2

The message is misleading as you're below the minimum required version.

**Comment by shcheklein** [comment]
> The message is misleading as you're below the minimum required version.

Yes, that's what I meant - it's a bug in the message. `Not found` -> 2.54.0 and a path to it, + regular button to configure, etc.

**Comment by shcheklein** [comment]
A few more things:

<img width="503" alt="Screenshot 2023-04-28 at 1 40 51 PM" src="https://user-images.githubusercontent.com/3659196/235252334-948910a6-52c2-4b75-a4e3-70fe384db671.png">

I don't think we need 3.0.0 (it's more of a safeguard, there is not 3.0 yet)

<img width="915" alt="Screenshot 2023-04-28 at 1 53 09 PM" src="https://user-images.githubusercontent.com/3659196/235252404-9571320e-635e-42ce-89b1-43e581cf9508.png">

Seems like a lot of empty space there w/o a reson?

**Comment by SoyGema** [comment]
Hey! First and foremost congratulations for keeping an eye on this. 
I'm reading the issue end to end and as I can see it starts super high-level and then goes into implementation. I'm trying to follow your Schema here.

* Regarding [high level ](https://github.com/iterative/vscode-dvc/issues/3434#issue-1616310739 )Proposing a **unique onboarding screen** for onboarding is a great idea and keeps the user-focused. As far as my exploration with other users during the workshop, the DVC Setup setup seems to be one of the main roadblocks. 

* The communication user/extension could be _more consistent in details, proactive and guided_ . Example:  See this snapshot in the Docker extension. Docker seems to have the same aspirations as your product: to deploy a container **without leaving the IDE**. However, it guides proactively by making questions and with a pop-up in the **walkthrough screen** (see image 1). Also, a pop-up appears (not in the image) to set up Docker from the Desktop, necessary step to work with the extension. Note that this is also in the Docker walkthrough, but they propose a **fastest path** by making two questions and showing a pop-up.  Please, if you decide to make the Setup on another screen, make sure the user can get there in less than 5 seconds without going necessarily to the Walkthrough.

<img width="1872" alt="Docker" src="https://user-images.githubusercontent.com/24204714/235681792-18acf098-c62d-4713-91bd-235bff4aa2c0.png">

Questions are super visible at first sight. Now, in your extension, you make a question and then a passive statement of DVC not configured without the warning sign. **It is much less visible at first sight for the user.** Would it be more consistent to make two questions and highlight them as Docker does? Minor. Is it necessary to be the first button to go to the walkthrough? Is not the user already seeing it ?? You are giving such an important place to something that is already there... Proposing Setup DVC verbose change . If not for this is there a way to do something like change the button colors ? how can you guys make this more visible?


<img width="1879" alt="GuideDVC" src="https://user-images.githubusercontent.com/24204714/235686500-da5e4044-d83a-43bb-bc5e-e5b6895801f0.png">

There are two issues I'm also experiencing.
1. I install the extension, I press the buttons and nothing happens. I consider this counterintuitive
Is this my problem ? How can I troubleshoot this ? Please let me know if this should go on a separate ticket. 

https://user-images.githubusercontent.com/24204714/235693979-7cbe7a83-3c48-45ae-8d23-44157ecc0fce.mp4

2. Getting Started. TBH, Every data scientist knows MNIST. Not many of your getting started scenarios. With MNIST in mind, a data scientist can **transfer dataset knowledge** into running experiments, and see what they can get from the extension. Besides, plots with that dataset can show good feature capabilities. Please, note that I'm looking at this from the Data Scientist perspective, if you are targeting more data engineers or software engineers this is not very important. If you are targeting a data scientist, I shall think of **how to load the Getting Started faster** without leaving the IDE.  Now, do I have to clone all your repository and access the module for getting it ? Is there a easiest / fastest way to do this?



- Regarding the screenshot that you are using. Is there a way to directly put the commands to upgrade the version or even a button that opens the CLI and does it? Can you help the user to solve the problem without leaving the IDE? There is a ton of effort put there but feels yet too robotic. _People seem to stick when you give them the solutions to their problems, not only when you explain in detail that they have one._ 


<img width="1234" alt="guide with commands" src="https://user-images.githubusercontent.com/24204714/235683583-530bc538-70cd-4d7b-920b-e15f6e5cf8b2.png">

* If you decided to leave out the UI - (buttons to set up the WorkSpace ), can you please contemplate a detailed guided communication for the extension (like including commands ) this for the extension?


Minor, User feedback from the Extension. The communication user/extension could be _more consistent in details, proactive, and guided following product aspirations_  . From Google Code extension. You can easily open feature requests and give feedback without leaving the IDE. Seriously, every time I go to a different screen I have some kind of cognitive load or attention to other stuff... Helping the user maintain focus might be a good idea.

<img width="1882" alt="Not leaving IDE Issue" src="https://user-images.githubusercontent.com/24204714/235694617-f0f42506-d0cb-4916-836f-2d9901b5875d.png">

I'm giving the last talks about the extensions and it´s great to close the stuff trying to give a helping hand on this instead of tweeting! <3 Many big rocks have been already tracked and Im happy to see this stuff aging well.  
 
Feel free to dismiss whatever might be not useful for you. With this action, I mark an important feedback/contrib task as done, but please don´t hesitate to ask whatever thing I might not be explaining myself well. 

Keep up the good work and have a nice day!

**Comment by julieg18** [comment]
>  Add some indicator as to whether sections are optional/completed to section titles.

Worked on some ideas for including an indicator into section titles! 

One idea was to add an icon on the right of the title:

<img width="630" alt="image" src="https://user-images.githubusercontent.com/43496356/235978204-3c725480-036d-41a2-8215-d8424f19c60c.png">

Another was updating the tooltip icon to signify whether something was incomplete/complete. Though this way makes the fact that there is a tooltip less obvious:

<img width="582" alt="image" src="https://user-images.githubusercontent.com/43496356/235978717-7d2480d9-0909-4a0f-af69-64d0dd0868dd.png">

What do we think? Any other ideas?

**Comment by shcheklein** [comment]
On the DVC CLI section. When there is no virtual environment the screen is a bit confusing and dangerous:

<img width="562" alt="Screenshot 2023-05-11 at 3 17 57 PM" src="https://github.com/iterative/vscode-dvc/assets/3659196/013763c7-63d5-43ee-a21c-9f1515c6f71b">

1. Minor. Can we add `:` after the `Version` (and the other table titles) please.
2. It suggests to install it into a global (user? will it add `-u`?) Python env which is not a good practice. What does Jupyter extension do in this case? 
3. We need to have a way to change the Python env and make it clear that this env is what Python ext selected by default. Maybe also show a way to install DVC from binaries, etc. 

To describe it in a different way. I would never hit Install in this case - too dangerous. Configure button is also not very explanatory (Auto mode goes into selecting an env but it's not obvious). Would be nice to have a create env flow, or at least notify. Need to think about this.

**Comment by shcheklein** [comment]
Here is the "Create an env" flow in the Jupyter extension: (fails since I picked to install all the requirements, but idea behind create an env is clear):

https://github.com/iterative/vscode-dvc/assets/3659196/d0701056-82e9-4f42-b7fe-017fd90f1e99

**Comment by julieg18** [comment]
Reviewed the remaining steps in this issue, considering time, priority, etc., and put together some next steps. 

> - [ ] Remove extra spacing below "Setup Complete" section
> - [ ] Add a reference to the walkthrough to a new "Get Started" section on the page.
> - [ ] Add a reminder of how to use DVCLive in the Setup Complete step of the Experiments section.

These tasks seem straightforward and should be quick to implement. I'll get started on them in the order above ^

> - [ ] Review sections, making them more static (for example, setup steps should aim to have only one paragraph of text and one row of buttons)
> - [ ] Look into moving our walkthrough into "Get Started" section instead of just adding a reference to it

These tasks will take some time and discussion, but I think they are of higher priority in comparison to the rest. Will start these after the smaller tasks mentioned above. 

> - [ ] Add plots section ( https://github.com/iterative/vscode-dvc/issues/3434#issuecomment-1507698885 )
> - [ ] Improve our "DVC unavailable" section, making recommended steps to install DVC more clear (https://github.com/iterative/vscode-dvc/issues/3434#issuecomment-1544772389)

I think we should move the "plots section" task along with the "dvc installation" task to their own issues to make discussion on more concrete next steps easier (with over 10 tasks done already, this issue has gotten quite large 📜 ).

**Comment by shcheklein** [comment]
@julieg18 this plan sounds good to me.

**Comment by julieg18** [comment]
Working on the "Get Started" section and thought about some solutions to avoid circular references and to further consolidate our Setup and walkthrough. 

### 1. Add a more clear separation between the two pages
Keep our "Get Started" walkthrough focused solely on walking through the extension features and the Setup would be focused on getting the user started with DVC installation. To do this:

  * The "Install DVC" step in the walkthrough would be deleted and we just add a brief mention in the beginning that tells you that you need to install DVC first before going through the walkthrough. 
  * A user doesn't see the walkthrough first when they first install the extension, instead they would see Setup first. The "Get Started" section would be "disabled" saying you should install DVC first. User goes through steps in the DVC section, then can go to the walkthrough and learn about the extension. 
  * Update links accordingly

### 2. Move Walkthrough into Setup
Delete our "Get Started" walkthrough, showing a walkthrough inside of the Setup "Get Started" section. To do this:

* Design a way to show a walkthrough inside the section. We'd most likely would need to simplify our walkthrough steps since we now have less space to show images/text. 
* Delete walkthrough entirely
* Update links accordingly

Personally, I like the first option more since we don't need to redesign/rewrite our walkthrough, but if we want our walkthrough simplified anyway and everything within "Setup", the second option would be the better way to go. Or maybe someone else has a better idea in mind :)

 I'll start working on the first option for now though!

**Comment by shcheklein** [comment]
Yes, I also think the first option is better and it makes sense to me to try to make it more and more like a features overview.
